### PR TITLE
lot57 rental property/ room features

### PR DESCRIPTION
**Pull request by schardtbc**
for review

### CODE REVIEW

File: src/graphql/rental-unit/rental-unit.resolver.ts [hunk 1]
```diff
@@ -30,11 +30,10 @@ export class RentalUnitResolver {
       @Context('reqUser') reqUser: ReqUser,
       ): Promise<CreateRentalUnitOutput> {
         const authUser = await UserEntity.findOne({userKey: reqUser.userKey});
-        const rentalProperty = await RentalPropertyEntity.findOne({rentalPropertyKey: input.rentalPropertyKey});
-
+        const managementCompanyKey = await RentalPropertyEntity.getManagementCompanyKey(input.rentalPropertyKey);
```
Comment (brianschardt) [inline src/graphql/rental-unit/rental-unit.resolver.ts]: This should be a method on the instance, should not be a static method.

File: src/shared/entities/rental-property.entity.ts [hunk 2]
```diff
@@ -107,4 +104,14 @@ export class RentalPropertyEntity extends BaseEntity {
     @ManyToMany(() => AmenityEntity)
     @JoinTable({name: 'AmenityXRentalProperty'})
     amenities: AmenityEntity[];
+
+    public static async getManagementCompanyKey( rentalPropertyKey: number): Promise<number> {
```
Comment (brianschardt) [inline src/shared/entities/rental-property.entity.ts]: Should not be static

File: src/graphql/rental-unit/rental-unit.resolver.ts (hunk 1 above)
Comment (schardtbc) [reply src/graphql/rental-unit/rental-unit.resolver.ts]: if I have an instance of the RentalProperty, I do not need a function to get the managementCompany as it is a property of the RentalPropertyEntity.

I would use the static function if all I have is the rentalPropertyKey, and I want to more efficiently just get the managmentCompanyKey.

File: src/shared/entities/rental-property.entity.ts (hunk 2 above)
Comment (schardtbc) [reply src/shared/entities/rental-property.entity.ts]: see prior comment

**Pull request closed without merging**